<a href="https://colab.research.google.com/github/amyahespanol/Amyah-Espanol-Statistical-Theory-APM1111/blob/main/Stats_SA1_Espanol.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import pandas as pd
import numpy as np
from scipy import stats

#Data Input
males = [12, 4, 11, 13, 11, 7, 9, 10, 10, 7, 7, 12, 6, 9, 15, 10, 11, 12, 7, 8,
         8, 9, 11, 10, 9, 10, 9, 9, 7, 9, 11, 7, 10, 10, 11, 9, 12, 12, 8, 13,
         9, 10, 8, 11, 10, 13, 13, 9, 10, 13]

females = [11, 9, 7, 10, 9, 10, 10, 7, 9, 10, 11, 8, 9, 6, 11, 10, 7, 9, 12, 14,
           11, 12, 12, 8, 12, 12, 9, 10, 11, 7, 12, 7, 9, 8, 11, 10, 8, 13, 8, 10,
           9, 9, 9, 11, 9, 9, 8, 9, 12, 11]

df_all = pd.concat([pd.DataFrame({'Hours': males, 'Gender': 'Male'}),
                    pd.DataFrame({'Hours': females, 'Gender': 'Female'})])

# Statistical Function
def calculate_formatted_stats(data):
    # Calculate basic descriptives and deciles
    percentiles = [0.1, 0.2, 0.25, 0.3, 0.4, 0.5, 0.6, 0.7, 0.75, 0.8, 0.9]
    s = data.describe(percentiles=percentiles)

    s.rename({'count': 'sample_size (n)'}, inplace=True)
    s['variance'] = data.var()
    s['skewness'] = data.skew()
    s['kurtosis'] = data.kurtosis()

    # 95% Confidence Interval
    mean, sem = data.mean(), stats.sem(data)
    ci = stats.t.interval(0.95, len(data)-1, loc=mean, scale=sem)
    s['95% CI Lower'] = ci[0]
    s['95% CI Upper'] = ci[1]

    #put sample size at the top
    items = list(s.index)
    items.insert(0, items.pop(items.index('sample_size (n)')))
    return s.reindex(items)

# Q1
print("--- I.1: ENTIRE DATASET SUMMARY ---")
overall = calculate_formatted_stats(df_all['Hours'])
# Use .to_string() to remove the "Name/dtype" line at the bottom
print(overall.to_string())

print("\n" + "="*40 + "\n")

# Q2
print("--- I.2: SUMMARY BY GENDER ---")
gender_grouped = df_all.groupby('Gender')['Hours'].apply(calculate_formatted_stats).unstack().T
print(gender_grouped.to_string())


--- I.1: ENTIRE DATASET SUMMARY ---
sample_size (n)    100.000000
mean                 9.760000
std                  1.964945
min                  4.000000
10%                  7.000000
20%                  8.000000
25%                  9.000000
30%                  9.000000
40%                  9.000000
50%                 10.000000
60%                 10.000000
70%                 11.000000
75%                 11.000000
80%                 11.200000
90%                 12.000000
max                 15.000000
variance             3.861010
skewness            -0.007343
kurtosis             0.015803
95% CI Lower         9.370112
95% CI Upper        10.149888


--- I.2: SUMMARY BY GENDER ---
Gender              Female       Male
sample_size (n)  50.000000  50.000000
mean              9.700000   9.820000
std               1.775686   2.154161
min               6.000000   4.000000
10%               7.000000   7.000000
20%               8.000000   8.000000
25%               9.000000   9.0000

# I.3 Statistical Report
The study analyzed cell phone usage for 100 college students (sample size (n) = 100). The mean (average) time spent talking on cell phones were 9.76 hours per week. The median was 10 hours. The data showed a narrow spread. It has a standard deviation of 1.96 and a variance of 3.86.The distribution is very balanced with a skewness of -0.007. This tells us the data is almost perfectly symmetrical. Kurtosis is 0.015. This means the distribution is close to a normal bell curve with no extreme outliers. Range was from a minimum of 4 hours to a maximum of 15 hours.

When grouping the data by gender (sample size (n) = 50 for each), the statistics were very similar. The males averaged (mean) 9.82 hours, and the females averaged less at 9.70 hours. This is a small difference though of 0.12 hours per week. Males had a little more variability with a standard deviation of 2.15 while the females' had an SD of 1.78. When we look at maximums and minimums, the highest usage was recorded by a male (15 hours), and the lowest was recorded by a male (4 hours). This tells us there might be more variance in the male group dataset.

The 95% Confidence Interval (CI) gives us the range we are 95% sure the population mean is. The overall CI is [9.37, 10.15] hours. The male CI is [9.21, 10.43] hours, and the female CI is [9.19, 10.21] hours. Because the confidence intervals for males and females are almost the same, we can say there is not a significant difference in cell phone usage between the two genders at this university statistically.

The pattern seen mostly is similarity between the groups. Both genders share the same median (10 hours) and very similar decile distributions. The data tells us gender does not strongly influence cell phone talk time at Midwestern University. Both groups have a normal distribution with a center of about 9–10 hours per week. The higher variance in males shows there are more high and low times among men, while female calls are more around the mean.


In [ ]:
import numpy as np

#Data
data = np.array([2, 3, 7, 8, 10, 10, 12, 15, 17, 20])
n = len(data)

# Raw Moments
m1_p = np.mean(data**1)
m2_p = np.mean(data**2)
m3_p = np.mean(data**3)
m4_p = np.mean(data**4)

print("--- II.1: Raw Moments ---")
print(f"(a) m1' = {m1_p}")
print(f"(b) m2' = {m2_p}")
print(f"(c) m3' = {m3_p}")
print(f"(d) m4' = {m4_p}")

#Central Moments
mean = np.mean(data)
m1 = np.mean((data - mean)**1)
m2 = np.mean((data - mean)**2)
m3 = np.mean((data - mean)**3)
m4 = np.mean((data - mean)**4)

print("\n--- II.2: Central Moments ---")
print(f"(a) m1 = {round(m1, 10)}") #0
print(f"(b) m2 = {m2}")
print(f"(c) m3 = {m3}")
print(f"(d) m4 = {m4}")

# Verify
rhs = m4_p - 4*m1_p*m3_p + 6*(m1_p**2)*m2_p - 3*(m1_p**4)
print("\n--- II.3: Verify Problem ---")
print(f"Calculate: {m4}")
print(f"m4_p - 4*m1_p*m3_p + 6*(m1_p**2)*m2_p - 3*(m1_p**4) result: {rhs}")
print(f"Equality: {np.isclose(m4, rhs)}")

--- II.1: Raw Moments ---
(a) m1' = 10.4
(b) m2' = 138.4
(c) m3' = 2090.6
(d) m4' = 34147.6

--- II.2: Central Moments ---
(a) m1 = -0.0
(b) m2 = 30.24
(c) m3 = 22.24799999999998
(d) m4 = 1898.9472

--- II.3: Verify Problem ---
Calculate: 1898.9472
m4_p - 4*m1_p*m3_p + 6*(m1_p**2)*m2_p - 3*(m1_p**4) result: 1898.9472000000023
Equality: True


# Part III: Mathematical Proof
**Prove that $m'_4 = m_4 + 4hm_3 + 6h^2m_2 + h^4$, given $h = m'_1$**

Define $m'_4 = E[X^4]$ (4th raw moment)

$m_k = E[(X - \mu)^k]$ (central moment)

$h = \mu$ (mean)

Expansion

Express $X$ as $(X - \mu) + \mu$. Substitute $h$ for $\mu$

$$m'_4 = E[((X - \mu) + h)^4]$$

Use Binomial Expansion

Let $a = (X - \mu)$ and $b = h$

$(a + b)^4 = a^4 + 4a^3b + 6a^2b^2 + 4ab^3 + b^4$

Then, $$m'_4 = E[(X - \mu)^4 + 4h(X - \mu)^3 + 6h^2(X - \mu)^2 + 4h^3(X - \mu) + h^4]$$

Distribute the Expectation $E[\cdot]$ to each term:

$$m'_4 = E[(X - \mu)^4] + 4hE[(X - \mu)^3] + 6h^2E[(X - \mu)^2] + 4h^3E[(X - \mu)] + E[h^4]$$

Simplify

$E[(X - \mu)^4] = m_4$$E[(X - \mu)^3] = m_3$$E[(X - \mu)^2] = m_2$$E[(X - \mu)] = 0$

(first central moment is always zero)

$E[h^4] = h^4$ (The expectation of a constant is the constant)

Final Result

Substitute back into the equation:

$$m'_4 = m_4 + 4h(m_3) + 6h^2(m_2) + 4h^3(0) + h^4$$

Therefore, $$m'_4 = m_4 + 4hm_3 + 6h^2m_2 + h^4$$

In [2]:
from scipy import stats

# Condition: Area between -t1 and t1 is 0.90, v = 20
# (1 - 0.90 = 0.10 -> 0.10 / 2 = 0.05)
t1_q1 = stats.t.ppf(1 - 0.05, df=20)

# Condition: Area to the left of -t1 is 0.025, v = 20
# find the value for 0.025 percentile
t1_q2 = stats.t.ppf(0.025, df=20)
t1_q2_abs = abs(t1_q2)

# 3. Combined area to the right of t1 and left of -t1 is 0.01, v = 5
# This means 0.005 is in each tail
t1_q3 = stats.t.ppf(1 - 0.005, df=5)

# 4. Area to the right of t1 is 0.55, v = 16
# This means the area to the left is 0.45
t1_q4 = stats.t.ppf(0.45, df=16)

print(f"Q1. t1 = {t1_q1:.3f}")
print(f"Q2. t1 = {t1_q2_abs:.3f}")
print(f"Q3. t1 = {t1_q3:.3f}")
print(f"Q4. t1 = {t1_q4:.3f}")

1. t1 = 1.725
2. t1 = 2.086
3. t1 = 4.032
4. t1 = -0.128
